tsdoc/src/nodes/DocNodeContainer.ts

```diff
@@ -36,24 +36,14 @@ export abstract class DocNodeContainer extends DocNode {
     return this._nodes;
   }
 
-  /**
-   * Returns true if the specified `docNode` is allowed to be added as a child node.
-   * The {@link appendNode()} and {@link appendNodes()} functions use this to validate their
-   * inputs.
-   *
-   * @virtual
-   */
-  public isAllowedChildNode(docNode: DocNode): boolean {
-    return false;
-  }
-
   /**
    * Append a node to the container.
    */
   public appendNode(docNode: DocNode): void {
-    if (!this.isAllowedChildNode(docNode)) {
-      throw new Error(`A ${this.kind} cannot contain nodes of type ${docNode.kind}`);
+    if (!this.configuration.docNodeManager.isAllowedChild(this.kind, docNode.kind)) {
```

Comment:
>configuration.docNodeManager.isAllowedChild [](start = 14, length = 43)

This is pretty long. Maybe we should alias `isAllowedChild` on the `DocNode` base class.

Reply:
Let's wait and see if any other code actually needs to do this

---
In reply to: [230253352](https://github.com/Microsoft/tsdoc/pull/126#discussion_r230253352) [](ancestors = 230253352)